docs/conf.py

```diff
@@ -200,7 +200,12 @@
 
 # Allow linking objects on other Sphinx sites seamlessly:
 intersphinx_mapping.update(
-    python=('https://docs.python.org/3', None),
+    # python=('https://docs.python.org/3', None),
+    python=('https://docs.python.org/3.11/', None),
+    # ^-- Python 3.11 is required because it still contains `distutils`.
+    #     Just leaving it as `3` would imply 3.12+, but that causes an
+    #     error with the cross references to disutils functions.
+    #     Inventory cache may cause errors, deleting it solves the problem.
```

Comment:
This workaround is fine as a stop gap. I think the long-term strategy is for Setuptools to own the distutils documentation (https://setuptools.pypa.io/en/latest/deprecated/distutils/index.html). Probably the proper fix is to simply re-point distutils references to this documentation. I filed #4081 to track.